modules/fx/arrow-fx/src/main/kotlin/arrow/fx/typeclasses/Bracket.kt

```diff
@@ -159,4 +159,15 @@ interface Bracket<F, E> : MonadError<F, E> {
         else -> just<Unit>(Unit)
       }
     }
+
+  /**
+   * Executes the given `finalizer` when the source is finished in error.
+   */
+  fun <A> Kind<F, A>.onError(finalizer: Kind<F, Unit>): Kind<F, A> =
```

Comment:
I think it might be better to pass the error `E` to the finalizer:
```
  fun <A> Kind<F, A>.onError(finalizer: (E) -> Kind<F, Unit>): Kind<F, A> =
```

Reply:
with this signature i get a runtime error i cannot figure out:

```
arrow.fx.extensions.io.concurrent.IODefaultConcurrentKt$concurrent_singleton$1.onError2(Larrow/Kind;Lkotlin/jvm/functions/Function1;)Larrow/Kind;
java.lang.AbstractMethodError: arrow.fx.extensions.io.concurrent.IODefaultConcurrentKt$concurrent_singleton$1.onError2(Larrow/Kind;Lkotlin/jvm/functions/Function1;)Larrow/Kind;
	at arrow.fx.typeclasses.ConcurrentContinuation.onError2(ConcurrentContinuation.kt)
	at arrow.fx.IOTest$46$1.invokeSuspend(IOTest.kt:566)

```

note: the function is named `onError2` in the stacktrace.